diff -ur a/docs/labwc-actions.5.scd b/docs/labwc-actions.5.scd
--- a/docs/labwc-actions.5.scd	2024-09-04 04:05:41.878247713 -0500
+++ b/docs/labwc-actions.5.scd	2024-09-04 04:10:56.494285744 -0500
@@ -134,6 +134,18 @@
 	windows that have client side decorations if it would increase the
 	amount of decoration. Default is no.
 
+*<action name="Decorate" />*
+	Similar to ToggleDecorations below but a single state
+	- Titlebar and borders
+
+*<action name="Undecorate" />*
+	Similar to ToggleDecorations below but a single state
+	- No titlebar or borders
+
+*<action name="BorderOnly" />*
+	Similar to ToggleDecorations below but a single state
+	- Borders only
+
 *<action name="ToggleDecorations" />*
 	Toggle decorations of focused window.
 
diff -ur a/src/action.c b/src/action.c
--- a/src/action.c	2024-09-04 04:05:41.883247841 -0500
+++ b/src/action.c	2024-09-04 04:10:56.494285744 -0500
@@ -118,7 +118,10 @@
 	ACTION_TYPE_TOGGLE_TABLET_MOUSE_EMULATION,
 	ACTION_TYPE_TOGGLE_MAGNIFY,
 	ACTION_TYPE_ZOOM_IN,
-	ACTION_TYPE_ZOOM_OUT
+	ACTION_TYPE_ZOOM_OUT,
+	ACTION_TYPE_DECORATE,
+	ACTION_TYPE_UNDECORATE,
+	ACTION_TYPE_BORDERONLY
 };
 
 const char *action_names[] = {
@@ -180,6 +183,9 @@
 	"ToggleMagnify",
 	"ZoomIn",
 	"ZoomOut",
+	"Decorate",
+	"Undecorate",
+	"BorderOnly",
 	NULL
 };
 
@@ -1167,6 +1173,21 @@
 		case ACTION_TYPE_ZOOM_OUT:
 			magnify_set_scale(server, MAGNIFY_DECREASE);
 			break;
+		case ACTION_TYPE_DECORATE:
+			if (view) {
+				view_set_ssd_mode(view, LAB_SSD_MODE_FULL);
+			}
+			break;
+		case ACTION_TYPE_UNDECORATE:
+			if (view) {
+				view_set_ssd_mode(view, LAB_SSD_MODE_NONE);
+			}
+			break;
+		case ACTION_TYPE_BORDERONLY:
+			if (view) {
+				view_set_ssd_mode(view, LAB_SSD_MODE_BORDER);
+			}
+			break;
 		case ACTION_TYPE_INVALID:
 			wlr_log(WLR_ERROR, "Not executing unknown action");
 			break;
