diff --git a/src/view.c b/src/view.c
index b1b4265..a2803ea 100644
--- a/src/view.c
+++ b/src/view.c
@@ -1436,12 +1436,18 @@ view_toggle_decorations(struct view *view)
 	assert(view);
 
 	enum ssd_mode mode = view_get_ssd_mode(view);
-	if (rc.ssd_keep_border && mode == LAB_SSD_MODE_FULL) {
-		view_set_ssd_mode(view, LAB_SSD_MODE_BORDER);
-	} else if (mode != LAB_SSD_MODE_NONE) {
-		view_set_ssd_mode(view, LAB_SSD_MODE_NONE);
+	if (rc.ssd_keep_border) {
+		if (mode == LAB_SSD_MODE_FULL) {
+			view_set_ssd_mode(view, LAB_SSD_MODE_BORDER);
+		} else {
+			view_set_ssd_mode(view, LAB_SSD_MODE_FULL);
+		}
 	} else {
-		view_set_ssd_mode(view, LAB_SSD_MODE_FULL);
+		if (mode == LAB_SSD_MODE_FULL) {
+			view_set_ssd_mode(view, LAB_SSD_MODE_NONE);
+		} else {
+			view_set_ssd_mode(view, LAB_SSD_MODE_FULL);
+		}
 	}
 }
 
